Source/WebCore/dom/Document.cpp

```diff
@@ -9710,10 +9710,13 @@ void Document::applyContentDispositionAttachmentSandbox()
     ASSERT(shouldEnforceContentDispositionAttachmentSandbox());
 
     setReferrerPolicy(ReferrerPolicy::NoReferrer);
-    if (!isMediaDocument())
-        enforceSandboxFlags(SandboxFlags::all());
-    else
+
+    if (isMediaDocument())
         enforceSandboxFlags(SandboxFlag::Origin);
+    else if (isPluginDocument())
```

Comment:
What kind of documents (or what kind of content) will return true for `isPluginDocument()`? You are changing the sandbox for all of them. Even if plugin documents are limited to PDFs today, are they guaranteed to be in the future?

Reply:
I doubt we will be reexpanding support for plugins past PDF (and in fact this assumption has already spread in the code; see e.g. WebKit2's PluginView knowing that it is only a PDF).

Reply:
It's highly unlikely that WebKit2's PluginView becomes non-exclusive to PDF content. At that point, we would have to revisit a lot of assumptions baked into our code.